### PR TITLE
Issue 7770 - add xref source annotation MONDO:otherHierarchy

### LINKED ISSUE #7770: Update docs for xref "MONDO:otherHierarchy"
When "MONDO:otherHierarchy" is added as a source value for a database cross reference it means that the external term Mondo is referencing is not in the "disease" branch. For example in this [PR](https://github.com/monarch-initiative/mondo/pull/7763#issuecomment-2142893384), [NCIT:C3307](https://www.ebi.ac.uk/ols4/ontologies/ncit/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FNCIT_C3307) is not in the disease branch of NCIT, but in the "Findings" branch.

Add this xref source annotation into https://mondo.readthedocs.io/en/stable/editors-guide/f-entities/#axiom-annotations-summary-table

### PR DESCRIPTION
**Pull request by twhetzel**
closes #7770 

Added annotation to: https://mondo.readthedocs.io/en/stable/editors-guide/f-entities/#axiom-annotations-summary-table

### CODE REVIEW

**Review by nicolevasilevsky — APPROVED** [review]

File: docs/editors-guide/mondo-docs-website-editing.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ Get the latest website files locally. This can either be done in the Terminal or
 1. Navigate to the root of the mondo repo, e.g. `cd YOUR-LOCAL-DIRECTORY/mondo`
 1. Fetch the latest content from `master`
 1. Create a feature branch using the GitHub issue number as the branch name, e.g. issue-1234
-1. Run the MkDocs server on your local computer by running this command in your Terminal at the root of the cloned mondo repo (i.e. at the same directory level where the `mkdocs.yml` file is located):  `mkdocs serve`. <br />To stop running the development server type Control+X Control+C in the Terminal.
+1. Run the MkDocs server on your local computer by running this command (`mkdocs serve`) in your Terminal at the root of the cloned mondo repo (i.e. at the same directory level where the `mkdocs.yml` file is located). <br />To stop running the development server type Control+X Control+C in the Terminal.
```
Comment (nicolevasilevsky) [inline docs/editors-guide/mondo-docs-website-editing.md:24]: @twhetzel were these changes intended on this PR

Comment (twhetzel) [reply docs/editors-guide/mondo-docs-website-editing.md:24]: yes
